src/appwindow.cpp

```diff
@@ -1328,6 +1332,13 @@ void AppWindow::onConfirmTriggered(MainWindow *widget)
         ui->tabWidget->setCurrentIndex(index);
 }
 
+void AppWindow::onWindowCloseRequested(MainWindow *window)
```

Comment:
To actually close the window with `:q`, mainwindow must be able to request appwindow to close itself.